ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -101,8 +107,18 @@ public void start() throws StoreException {
               "Failed to acquire lock on file " + dataDir + ". Another process or thread is using this directory.",
               StoreErrorCodes.Initialization_Error);
         }
+        File storeDescriptorFile = new File(dataDir, STORE_DESCRIPTOR);
+        StoreDescriptor storeDescriptor = null;
+        if (storeDescriptorFile.exists()) {
+          DataInputStream dataInputStream = new DataInputStream(new FileInputStream(storeDescriptorFile));
+          storeDescriptor = StoreDescriptor.fromBytes(dataInputStream);
+        } else {
+          storeDescriptor = new StoreDescriptor(storeId, UUID.randomUUID());
+          // TODO create and persist the file
+        }
         log = new Log(dataDir, capacityInBytes, config.storeSegmentSizeInBytes, metrics);
-        index = new PersistentIndex(dataDir, taskScheduler, log, config, factory, recovery, hardDelete, metrics, time);
+        index = new PersistentIndex(dataDir, taskScheduler, log, config, factory, recovery, hardDelete, metrics, time,
+            storeDescriptor.incarnationId);
```

Comment:
if you are abstracting this, please make this a function and not a directly accessible variable.